core/src/main/scala/org/apache/spark/deploy/mesos/MesosExternalShuffleService.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.mesos
+
+import java.util.concurrent.CountDownLatch
+
+import scala.collection.mutable
+
+import org.apache.spark.deploy.ExternalShuffleService
+import org.apache.spark.rpc.{RpcAddress, RpcEnv, ThreadSafeRpcEndpoint}
+import org.apache.spark.util.Utils
+import org.apache.spark.{Logging, SecurityManager, SparkConf}
+
+private[mesos] class MesosExternalShuffleService(conf: SparkConf, securityManager: SecurityManager)
+  extends ExternalShuffleService(conf, securityManager) {
+
+  private var rpcEnv: RpcEnv = _
+
+  override def start(): Unit = {
+    super.start()
+    val port = conf.getInt("spark.mesos.shuffle.service.port", 7327)
```

Comment:
It's actually a additional endpoint on a different port that is listening for the driver.

Let me take a look if I can easily manage to use the same, but it looks like it's not using the endpoint mechanism.

> On Aug 1, 2015, at 11:59 AM, andrewor14 <notifications@github.com> wrote:
> 
> In core/src/main/scala/org/apache/spark/deploy/mesos/MesosExternalShuffleService.scala:
> 
> > +
> > +import scala.collection.mutable
> > +
> > +import org.apache.spark.deploy.ExternalShuffleService
> > +import org.apache.spark.rpc.{RpcAddress, RpcEnv, ThreadSafeRpcEndpoint}
> > +import org.apache.spark.util.Utils
> > +import org.apache.spark.{Logging, SecurityManager, SparkConf}
> > +
> > +private[mesos] class MesosExternalShuffleService(conf: SparkConf, securityManager: SecurityManager)
> > +  extends ExternalShuffleService(conf, securityManager) {
> > +
> > +  private var rpcEnv: RpcEnv = _
> > +
> > +  override def start(): Unit = {
> > +    super.start()
> > +    val port = conf.getInt("spark.mesos.shuffle.service.port", 7327)
> why would you have both shuffle services on the same cluster? Your new shuffle service is a replacement for the old one, no?
> 
> —
> Reply to this email directly or view it on GitHub.
>